app/src/main/java/com/google/android/samples/socialite/ui/Main.kt

```diff
@@ -66,124 +76,160 @@ fun MainNavigation(
-                navArgument("text") { defaultValue = "" },
-            ),
-            deepLinks = listOf(
-                navDeepLink {
-                    action = Intent.ACTION_VIEW
-                    uriPattern = "https://socialite.google.com/chat/{chatId}"
-                },
-            ),
-        ) { backStackEntry ->
-            val chatId = backStackEntry.arguments?.getLong("chatId") ?: 0L
-            val text = backStackEntry.arguments?.getString("text")
-            ChatScreen(
-                chatId = chatId,
-                foreground = true,
-                onBackPressed = { navController.popBackStack() },
-                onCameraClick = { navController.navigate("chat/$chatId/camera") },
-                onPhotoPickerClick = { navController.navigateToPhotoPicker(chatId) },
-                onVideoClick = { uri -> navController.navigate("videoPlayer?uri=$uri") },
-                prefilledText = text,
-                modifier = Modifier.fillMaxSize(),
-            )
-        }
-        composable(
-            route = "chat/{chatId}/camera",
-            arguments = listOf(
-                navArgument("chatId") { type = NavType.LongType },
-            ),
-        ) { backStackEntry ->
-            val chatId = backStackEntry.arguments?.getLong("chatId") ?: 0L
-            Camera(
-                onMediaCaptured = { capturedMedia: Media? ->
-                    when (capturedMedia?.mediaType) {
-                        MediaType.PHOTO -> {
-                            navController.popBackStack()
-                        }
+            composable<Route.ChatsList> {
+                SocialiteNavSuite(
+                    navController,
+                    modifier = Modifier.sharedElement(
+                        rememberSharedContentState(key = "navsuite"),
+                        animatedVisibilityScope = this@composable,
+                    ),
+                ) {
+                    ChatList(
+                        onChatClicked = { chatId -> navController.navigate(Route.ChatThread(chatId)) },
+                        modifier = Modifier.fillMaxSize(),
+                    )
+                }
+            }
 
-                        MediaType.VIDEO -> {
-                            navController.navigate("videoEdit?uri=${capturedMedia.uri}&chatId=$chatId")
-                        }
+            composable<Route.Timeline> {
+                SocialiteNavSuite(
+                    navController,
+                    modifier = Modifier.sharedElement(
+                        rememberSharedContentState(key = "navsuite"),
+                        animatedVisibilityScope = this@composable,
+                    ),
+                ) {
+                    Timeline(Modifier.fillMaxSize())
+                }
+            }
+
+            composable<Route.Settings> {
+                SocialiteNavSuite(
+                    navController,
+                    modifier = Modifier.sharedElement(
```

Comment:
I think there's an interesting issue here when everything comes together with `Modifier.sharedElement` + `NavigationSuiteScaffold` inside a `NavHost`.

The `Modifier` here that is passed to `SocialiteNavSuite` ends up being applied to the navigation UI (as displayed by the `NavigationSuiteScaffold`) _and_ all of the contained content.

We don't want all of the contained content to be part of the shared element - the only thing that should be shared is the navigation UI itself. I think we can accomplish that by using the `NavigationSuiteScaffoldLayout` `@Composable` and `NavigationSuite` instead, which allows us to set the shared element `Modifier` on the navigation UI itself.